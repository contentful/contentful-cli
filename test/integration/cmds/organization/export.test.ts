import nixt from 'nixt'
import { join } from 'path'
const bin = join(__dirname, './../../../../', 'bin')
const { replaceCopyrightYear } = require('../../util')

const organizationId = process.env.CLI_E2E_ORG_ID
const org = process.env.CLI_E2E_ORG_ID

const app = () => {
  return nixt({ newlines: true }).cwd(bin).base('./contentful.js ').clone()
}

type Result = {
  stderr: string
  stdout: string
}

const cmd = 'organization export'

describe('organization export snapshots', () => {
  test('should print help message', done => {
    app()
      .run(`${cmd} --help`)
      .code(0)
      .expect(result => {
        const text = result.stdout.trim()
        const resultText = replaceCopyrightYear(text)
        expect(resultText).toMatchSnapshot('help data is correct')
      })
      .end(done)
  })
  it('should exit 1 when no args', done => {
    app()
      .run(`${cmd}`)
      .code(1)
      .expect(({ stderr }: Result) => {
        const resultText = stderr.trim()

        expect(resultText).toContain('Usage: contentful organization export')
      })
      .end(done)
  })
  it('should exit 1 when no organization id passed or in context', done => {
    app()
      .run(`${cmd} --header 'Authorization'`)
      .code(1)
      .expect(({ stderr }: Result) => {
        const resultText = stderr.trim()
        expect(resultText).toContain(
          'Missing required argument: organization-id'
        )
      })
      .end(done)
  })
  it('should return concepts / concepts scheme', done => {
    app()
      .run(`${cmd} --organization-id ${organizationId} --save-file=false`)
      .code(0)
      .expect(({ stdout }: Result) => {
        const resultText = stdout.trim()

        expect(resultText).toContain('Exporting Concepts')
        expect(resultText).toContain('Exporting Concepts')
        expect(resultText).toContain('Exporting Concept Schemes')
        expect(resultText).toContain('Organization data exported to')
        expect(resultText).toContain('concepts')
        expect(resultText).toContain('conceptSchemes')
      })
      .end(done)
  })
  it('should suppress any log output when silent is true', done => {
    app()
      .run(`${cmd} --organization-id ${organizationId} --silent=true`)
      .code(0)
      .expect(({ stdout }: Result) => {
        const resultText = stdout.trim()

        expect(resultText).not.toContain('Exporting Concepts')
        expect(resultText).not.toContain('Exporting Concepts')
        expect(resultText).not.toContain('Exporting Concept Schemes')
        expect(resultText).not.toContain('Organization data exported to')
        expect(resultText).not.toContain('concepts')
        expect(resultText).not.toContain('conceptSchemes')
      })
      .end(done)
  })
})
