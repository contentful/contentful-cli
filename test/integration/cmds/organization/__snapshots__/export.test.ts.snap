// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organization export snapshots should print help message: help data is correct 1`] = `
"Usage: contentful organization export

Options:
  -h, --help                Show help                                  [boolean]
  --management-token, --mt  Contentful management API token             [string]
  --organization-id, --oid  ID of Organization with source data
                                                             [string] [required]
  --header, -H              Pass an additional HTTP Header              [string]
  --output-file, -o         Output file. It defaults to
                            ./migrations/<timestamp>-<organization-id>.json
                                                                        [string]

See more at:
https://github.com/contentful/contentful-cli/tree/master/docs/organization/expor
t
Copyright 2013 Contentful"
`;
